### PR TITLE
fix download URL

### LINKED ISSUE #335: Update download url
Currently in Gittyup we are using the following download url: https://github.com/Murmele/gittyup/releases/download/stable/Gittyup%1%2.%3

The problem is that we don't have anymore a stable release.

As a hot fix I changed the label of the latest development release to be the "stable" it is the same as gittyup_v1.2.0 but this was a dirty fix.

We have to change it to:
https://github.com/Murmele/gittyup/releases/download/v%1/Gittyup%2%3.%4

### PR DESCRIPTION
**Pull request by Murmele**
fixes #335

### CODE REVIEW

File: src/update/Updater.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ namespace {
 const QString kDownloadPlatform = "Github";
 const QString kTemplateFmt = "%1-XXXXXX.%2";
 const QString kLinkFmt = "https://github.com/Murmele/gittyup/releases/download/"
-                         "stable/Gittyup%1%2.%3";
+                         "latest/Gittyup%1%2.%3";
```
Comment (alerque) [inline src/update/Updater.cpp:50]: I don't think this is right, to use the GitHub release keywords the scheme needs to be `releases/latest/download` not `releases/download/latest`. Also this scheme should probably be used just to download the manifest, then the manifest used to extract the tag name and construct the actual download URL.

**Review by alerque — CHANGES_REQUESTED** [review]

File: src/update/Updater.cpp (hunk 1 above)
Comment (Murmele) [reply src/update/Updater.cpp:50]: https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases#linking-to-the-latest-release

Comment (alerque) [reply src/update/Updater.cpp]: The one I recommend (and the GH docs you link to recommends) is the **keyword** "latest" that automatically gets URL redirected to the actual tag that happens to be the most recent stable release. The one you used is a **tag** that you have to manually create and update and isn't actually the version tag but a separate one presumably on the same commit but a parallel universe to the actual versioned tags. You want to stop having to create sliding tags for things and only use tags with actual version numbers and use the GH release API to link to the current latest release, not a tag that just happens to have the name "latest".

**Comment by Murmele** [comment]
@exactly-one-kas  @alerque should we create a bug fix release after this MR

We still need a sliding tag, because the latest push to master cannot be retrieved from the github API.

File: docs/changelog.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
+### v1.2.1 - 2022-10-31
+
+Bug fix release
+
+#### Changed
+
+* Fix download url for flatpak and macos
+
```
Comment (Murmele) [inline docs/changelog.md:8]:
```suggestion
* Fix Segmentation fault when ignoring files
```

File: docs/changelog.md [hunk 3]
```diff
@@ -1,3 +1,13 @@
+### v1.2.1 - 2022-10-31
```
Comment (Murmele) [inline docs/changelog.md:1]:
```suggestion
### v1.2.1 - 2022-11-09
```

**Comment by alerque** [comment]
> @exactly-one-kas @alerque should we create a bug fix release after this MR

Yes, pending the additional fix suggested below.

> We still need a sliding tag, because the latest push to master cannot be retrieved from the github API.


Actually you can, but what you are doing right now is regularly attaching binary artifacts to the HEAD commit, and that is a bit more complicated.

You can keep a sliding tag if you really don't want to change the architecture, **but the tag should not  be named "latest" it should be "snapshot" or "development" or something like that**. I suggest fixing the code now to in the future look for a tag with the proper name for unstable snapshot builds so that there isn't future confusion between the "latest" keyword GitHub uses and a manual tag of the same name.

**Comment by Murmele** [comment]
for me **development** sounds fine. @exactly-one-kas do you agree with it?

**Comment by exactly-one-kas** [comment]
`development` is fine by me

File: docs/changelog.md [hunk 4]
```diff
@@ -1,3 +1,13 @@
+### v1.2.1 - 2022-11-09
```
Comment (Murmele) [inline docs/changelog.md:1]:
```suggestion
### v1.2.1 - 2022-11-10
```

**Comment by Murmele** [comment]
After merging we can cross check the urls again and if they are working I will set the tag. Then we have to create last time the stable release tag

**Review by alerque — APPROVED** [review]

**Review by alerque — APPROVED** [review]
Looks right to me now.

**Review by exactly-one-kas — APPROVED** [review]

**Pull request merged**